types/reference/image_reference_test.go

```diff
@@ -0,0 +1,72 @@
+package reference
+
+import (
+	"testing"
+)
+
+func TestParse(t *testing.T) {
+	testCases := []struct {
+		ref           string
+		expectedName  string
+		expectedTag   string
```

Comment:
I'm not sure we should reinforce the concept of a tag being a digest. Maybe call this `expectedTagOrDigest`.